src/App.tsx

```diff
@@ -1,61 +1,98 @@
+import { FormEvent, useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(todosFromServer);
+  const [userId, setUserId] = useState(0);
+  const [isError, setIsError] = useState(false);
+
+  const resetTodos = () => {
+    setTitle('');
+    setUserId(0);
+    setIsError(false);
+  };
+
+  const addTodo = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (userId === 0 || !title) {
+      setIsError(true);
+
+      return;
+    }
+
+    const newTodo = {
+      id: todos.length + 1,
+      title,
+      userId,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+    resetTodos();
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={addTodo}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          Title:
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={title}
+            onChange={(event) => setTitle(event.target.value)}
+          />
+          <span className="error">
+            {isError && !title ? 'Please enter a title' : ''}
+          </span>
```

Comment:
For what reason do you need to render empty string?
Fix all cases
```suggestion
          {isError && !title && (
             <span className="error">
               Please enter a title
            </span>
          )}
         
```